### PR TITLE
Add option to remove `(n)` suffix in `s_num_patients`

### LINKED ISSUE #805: Add option to remove `"(n)"` suffix from `s_num_patients` output
### Feature description

Allow the user to remove the `"(n)"` suffix from `unique_count` labels when default labels are used.

This has been requested several times by the chevron team.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines.

### Security Policy

- [X] I agree to follow this project's Security Policy.

### PR DESCRIPTION
**Pull request by edelarua**
Closes #805

### CODE REVIEW

File: R/summarize_num_patients.R [hunk 1]
```diff
@@ -32,11 +34,12 @@ NULL
 #'   .N_col = 6L,
 #'   count_by = as.character(c(1, 1, 2, 1, 1, 1))
 #' )
-s_num_patients <- function(x, labelstr, .N_col, count_by = NULL) { # nolint
+s_num_patients <- function(x, labelstr, .N_col, count_by = NULL, unique_count_suffix = TRUE) { # nolint
 
   checkmate::assert_string(labelstr)
   checkmate::assert_count(.N_col)
   checkmate::assert_multi_class(x, classes = c("factor", "character"))
+  checkmate::assert_logical(unique_count_suffix)
```
Comment (Melkiades) [inline R/summarize_num_patients.R:42]: maybe better `assert_flag`?

File: R/summarize_num_patients.R [hunk 2]
```diff
@@ -50,7 +53,7 @@ s_num_patients <- function(x, labelstr, .N_col, count_by = NULL) { # nolint
   out <- list(
     unique = formatters::with_label(c(count1, ifelse(count1 == 0 && .N_col == 0, 0, count1 / .N_col)), labelstr),
     nonunique = formatters::with_label(count2, labelstr),
-    unique_count = formatters::with_label(count1, paste(labelstr, "(n)"))
+    unique_count = formatters::with_label(count1, if (!unique_count_suffix) labelstr else paste(labelstr, "(n)"))
```
Comment (Melkiades) [inline R/summarize_num_patients.R:56]: Probably does not change a lot, but what about using the more compact: `ifelse(unique_count_suffix, labelstr, paste(labelstr, "(n)"))`?

**Review by Melkiades — CHANGES_REQUESTED** [review]
Well done, almost there. Should we add also a small test?

File: R/summarize_num_patients.R (hunk 2 above)
Comment (edelarua) [reply R/summarize_num_patients.R]: Done.

**Review by Melkiades — APPROVED** [review]
Lgtm!! Well done Emily ^^

**Pull request merged**